### DISCUSSION TITLE
custom breakpoints not working

### DISCUSSION THREAD

**Discussion by tristan-derez** [chakra-ui/chakra-ui#5222] [Q&A] [answered]

**Comment by primos63** [comment]
It should be
```javascript
export const theme = extendTheme({
    breakpoints,
```
`breakpoints: { breakpoints }`  creates a breakpoints object in a breakpoints object
![image](https://user-images.githubusercontent.com/2768849/145525139-727033b5-c201-4567-beb7-139caaa37230.png)
You also need to set the `2xl` key appropriately; otherwise, you will end up with this
![image](https://user-images.githubusercontent.com/2768849/145525421-a5624a96-d416-436b-aa27-34d7e654c51f.png)

**Reply by tristan-derez** [reply]
Thanks!

But as I did that i encountered other problems. 

```ts
const breakpoints = createBreakpoints({
    sm: "34em",
    md: "60em",
    lg: "76.8em",
    xl: "144em",
    "2xl": "256em",
});

export const theme = extendTheme({
    breakpoints,
```

at 320px things are working fine but on wider size until 768px width, my divs should be 327px wide and its not.
and breakpoints still seem to not works as things change on breakpoints i didnt specify

**Comment by primos63** [comment, marked as answer]
It is working correctly.

Your `breakpoints` are `{  sm: "34em", md: "60em", lg: "76.8em", xl: "144em", "2xl": "256em" }` or `{ sm: "544px", md: "960px", lg: "1229px", xl: "2304px", 2xl: "4096px" }`.
Your `Card` has a responsive width of ["30em", "32.7em", "57.3em", "73em"] or ["480px", "523px", "917px", "1168px"].

So `Card` component will have the following widths:
- 0 to 543px ======> Card width is 30em
- 544px to 959px ==> Card width is 32.7em
- 960px to 1228px => Card width is 57.3em
- 1229px and on ==>  Card width is 73em

Because of the monospaced font used your em will be at 10px not 16px. The big issue is the misuse of responsive widths for every component. You should probably be using a responsive padding instead. You also mix `em`s and `rem`s so that's going to make your sizes look weird.

The config object should be
```javascript
config: {
  initialColorMode: 'system',
  useSystemColorMode: false
 }
 ```
 These settings are documented [here](https://chakra-ui.com/docs/features/color-mode#behavior-of-colormode).
 
I'm not going to correct everything, but [this](https://codesandbox.io/s/lingering-wind-wbl7h?file=/src/theme.ts) may be closer to what you're expecting.

**Reply by tristan-derez** [reply]
Oh, my bad, i'll just use px then. 

Thank you for taking the time to answer my questions :smile:

About the config file, if I do 

```ts 
config: {
  initialColorMode: 'system',
  useSystemColorMode: false
 }
```

I get the error message:

```
Type '"system"' is not assignable to type 'ColorMode'.
```

**Reply by primos63** [reply]
If you haven't upgraded @chakra-ui/react from 1.6.5, I would do so. The `system` setting is fairly new.
